src/Symfony/Component/Serializer/Serializer.php

```diff
@@ -256,6 +257,24 @@ private function getNormalizer(mixed $data, ?string $format, array $context): ?N
                     continue;
                 }
 
+                if ($normalizer instanceof SupportedTypesMethodInterface && ($supportedTypes = $normalizer->getSupportedTypes())) {
```

Comment:
we should compare to null, so that returning the empty array isn't the same as returning null:
`if ($normalizer instanceof SupportedTypesMethodInterface && null !== $supportedTypes = $normalizer->getSupportedTypes()) {`